packages/modules/providers/cache-redis/src/services/redis-cache.ts

```diff
@@ -0,0 +1,276 @@
+import { Redis } from "ioredis"
+import { RedisCacheModuleOptions } from "@types"
+
+export class RedisCachingProvider {
+  static identifier = "cache-redis"
+
+  protected redisClient: Redis
+  protected keyNamePrefix: string
+  protected defaultTTL: number
+
+  constructor(
+    { redisClient, prefix }: { redisClient: Redis; prefix: string },
+    options?: RedisCacheModuleOptions
+  ) {
+    this.redisClient = redisClient
+    this.keyNamePrefix = prefix
+    this.defaultTTL = options?.ttl ?? 3600 // 1 hour default
+  }
+
+  #getKeyName(key: string): string {
+    return `${this.keyNamePrefix}${key}`
+  }
+
+  #getTagKey(tag: string): string {
+    return `${this.keyNamePrefix}tag:${tag}`
+  }
+
+  #getOptionsKey(key: string): string {
+    return `${this.keyNamePrefix}options:${key}`
+  }
+
+  async get({ key, tags }: { key?: string; tags?: string[] }): Promise<any> {
+    if (key) {
```

Comment:
self-review: as carlos mention, we might need to store in memory the pair key/tags-promise and return the same promise for all callers